app/components/collections/show/deposits_component.rb

```diff
@@ -22,31 +22,22 @@ def collection_deposits
       end
 
       def values_for(work)
+        presenter = work_presenter_for(work)
         [
           helpers.link_to(work.title, work_or_wait_path(work)),
           work.user.name,
-          status_for(work),
+          presenter.status_message,
           work.object_updated_at ? helpers.l(work.object_updated_at, format: '%b %d, %Y') : nil,
-          persistent_link_for(work)
+          presenter.sharing_link
```

Comment:
I appreciate moving more of this logic into the presenter objects once we discover it's not component-specific.